react/features/toolbox/components/native/VideoMuteButton.tsx

```diff
@@ -1,7 +1,31 @@
 import { connect } from 'react-redux';
 
+import { VIDEO_MUTED_CHANGED } from '../../../base/conference/actionTypes';
 import { translate } from '../../../base/i18n/functions';
 import AbstractVideoMuteButton, { IProps, mapStateToProps } from '../AbstractVideoMuteButton';
 
+/**
+ * Component that renders native toolbar button for toggling video mute.
+ *
+ * @augments AbstractVideoMuteButton
+ */
+class VideoMuteButton extends AbstractVideoMuteButton<IProps> {
+    /**
+     * Changes video muted state and dispatches the state to redux.
+     *
+     * @override
+     * @param {boolean} videoMuted - Whether video should be muted or not.
+     * @protected
+     * @returns {void}
+     */
+    _setVideoMuted(videoMuted: boolean) {
+        this.props.dispatch?.({
+            type: VIDEO_MUTED_CHANGED,
```

Comment:
Why do we need new actions if SET_XXX_MUTED already exists?

Reply:
I think `SET_XXX_MUTED` are doing more stuff in reducers so not sure if dispatching them here wouldn't cause any undesired behaviour. Currently `XXX_MUTED_CHANGED` actions only have RN SDK handling for custom listeners.

That being said, it's ok to re-use `SET_XXX_MUTED` if no other reducer will catch those actions or cause side effects.

Reply:
They are already dispatched when you use the button, in the common code. These new actions are unnecessary.

Reply:
I see! I'll send a PR with refactoring later today then.

Reply:
Thank you!